### DISCUSSION TITLE
Payments, check successful_payment, help!!!

### DISCUSSION THREAD

**Discussion by akula222** [DinoLeung/TeleDart#261] [Q&A]
it's work
```
teledart.onPreCheckoutQuery().listen((event) async {
    await teledart.answerPreCheckoutQuery(event.id, true);
  });
```
how to check successful_payment ?
```
  teledart.onMessage(entityType: 'successful_payment', keyword: 'successful_payment').listen((event) async {
    print(event.successful_payment?.order_info);
    await event.reply("++++");
   });
```
It doesn’t work like that, help me how to do it right?

**Comment by akula222** [comment]
help me!!! @gladimdim @f3ath @simolus3 @brhanem

**Comment by The0xArchitect** [comment]
I don't think there's a way to add specific listener for the `successful_payment` service messages as of now.

But, you can do try this snippet:

```dart
  bot.onMessage().listen((msg) {
    if (msg.successfulPayment == null) return;

    final successfulPayment = msg.successfulPayment!;
    // Now do your stuff with successfulPayment here
  });
```

As you can see, this snippet will listen to all the messages, and execute the code only if `Message.successfulPayment` is not null.
Hope this helps.
